### PR TITLE
OTA privacy enhancement

### PR DESCRIPTION
**Pull request by flatsiedatsie**
Adds the option to disallow device-initiated OTA update checks.

Currently these OTA update checks can be quite frequent, and essentially mean the device is indirectly able to ping a cloud server. In the EU it's wise to handle an IP address as personal information, and therefore it might be wise to not enable OTA checking by default. Ideally a user would agree to a privacy policy that clarifies things a bit first.

### CODE REVIEW

**Comment by Koenkk** [comment]
Thanks for the PR. Few points that needs to be fixed for merge:
- Add a test case for this, https://github.com/Koenkk/zigbee2mqtt/blob/d1550c7511d4984b04bdb8e3136069abbd80fa32/test/otaUpdate.test.js#L218 can be used as a base
- Add a default value in `defaults`: https://github.com/Koenkk/zigbee2mqtt/blob/dev/lib/util/settings.js
- Add option to schema: https://github.com/Koenkk/zigbee2mqtt/blob/dev/lib/util/settings.schema.json
- Create a PR to update the following docs: https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/information/ota_updates.md
- Put option under `advanced` (not `ota`) `advanced.disable_automatic_ota_update_check`

**Comment by timoline** [comment]
Put option under advanced (not ota) advanced.disable_automatic_ota_update_check

hmmm, you don't want to add new otions to the correct section already ?
(Will be less work for later (when reorganising them)

**Comment by Koenkk** [comment]
@timoline agree, lets put it under `ota. disable_automatic_ota_update_check`.

File: lib/util/settings.js [hunk 1]
```diff
@@ -137,6 +137,17 @@ const defaults = {
...
     },
+    ota:{
+        /**
+         * Minimal time delta in milliseconds between polling third party server for potential firmware updates
+         */
+        update_check_interval: 10,
+        /**
+         * Completely disallow Zigbee devices to initiate a search for a potential firmware update. If set to true, only a user-initiated update search will be possible.
+         */
+        disable_automatic_update_check: false,
```
Comment (Koenkk) [inline lib/util/settings.js:148]: What are the uses cases for having both options, wouldn't `disable_automatic_update_check` already be enough?

**Comment by flatsiedatsie** [comment]
I have done everything on the list, except add a test case. I'm a novice programmer, and don't know how to do this. Could someone help me out here?

File: lib/util/settings.js (hunk 1 above)
Comment (flatsiedatsie) [reply lib/util/settings.js:148]: I personally would be very happy with `disable_automatic_update_check` only.

But I figured some people would be ok with device-initiated checking as long as they had control over the interval. Checking every 10 minutes can still result i quite a lot of connections to third party servers.

**Comment by Koenkk** [comment]
Thanks!

**Comment by flatsiedatsie** [comment]
My pleasure!

**Pull request merged**